src/openrct2/management/Research.cpp

```diff
@@ -601,20 +601,29 @@ void ride_entry_set_invented(int32_t rideEntryIndex)
 
 bool scenery_is_invented(const ScenerySelection& sceneryItem)
 {
-    assert(sceneryItem.SceneryType < SCENERY_TYPE_COUNT);
-    return _researchedSceneryItems[sceneryItem.SceneryType][sceneryItem.EntryIndex];
+    if (sceneryItem.SceneryType < SCENERY_TYPE_COUNT)
+    {
+        return _researchedSceneryItems[sceneryItem.SceneryType][sceneryItem.EntryIndex];
+    }
+    else
+    {
+        Guard::Assert(false, "Invalid Scenery Type");
+        return false;
+    }
 }
 
 void scenery_set_invented(const ScenerySelection& sceneryItem)
 {
-    assert(sceneryItem.SceneryType < SCENERY_TYPE_COUNT);
-    _researchedSceneryItems[sceneryItem.SceneryType][sceneryItem.EntryIndex] = true;
+    Guard::Assert(sceneryItem.SceneryType < SCENERY_TYPE_COUNT, "Invalid Scenery Type");
```

Comment:
I've always viewed Guard::Assert as an exit point of the program. I realise that isn't the case if you hit ignore but should we really be continuing. Wouldn't it be better to just `log_warning`

Reply:
Sure, I'd ditch the ignore anyway, if we could. In any case this does seem a bit out of the scope of this PR, no?

Reply:
I prefer replacing Guard::Assert with `log_warning` as well.

Reply:
Ok, done